partiql-planner/src/main/kotlin/org/partiql/planner/typer/PlanTyper.kt

```diff
@@ -917,4 +969,25 @@ internal class PlanTyper(
     private fun StructType.withNullableFields(): StructType {
         return copy(fields.map { it.copy(value = it.value.asNullable()) })
     }
+
+    private fun excludeBindings(input: List<Rel.Binding>, item: Rel.Op.Exclude.Item): List<Rel.Binding> {
+        var matchedRoot = false
+        val output = input.map {
+            if (item.root.isEquivalentTo(it.name)) {
+                matchedRoot = true
+                // recompute the StaticType of this binding after apply the exclusions
+                val type = it.type.exclude(item.steps, false)
+                it.copy(type = type)
+            } else {
+                it
+            }
+        }
+        if (!matchedRoot) handleUnresolvedExcludeRoot(item.root.symbol)
+        return output
+    }
+
+    private fun Identifier.Symbol.isEquivalentTo(other: String): Boolean = when (caseSensitivity) {
+        Identifier.CaseSensitivity.SENSITIVE -> symbol.equals(other)
+        Identifier.CaseSensitivity.INSENSITIVE -> symbol.equals(other, ignoreCase = true)
+    }
```

Comment:
Any reason to have `Identifier.Symbol.isEquivalentTo(other: String)` in both this file and `TypeUtils.kt`?

Reply:
This extension could be public in `org.partiql.plan` actually. But I put it here to keep it private

Reply:
I don't mind minor duplication of logic when it's just syntax sugar and lets use keep things private. Internal visibility in Kotlin is Java public